File: app/src/main/java/com/example/administrator/wuanandroid/ui/GroupingActivity.java
package com.example.administrator.wuanandroid.ui;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.example.administrator.wuanandroid.Bean.GroupBean.GroupingResponse;
import com.example.administrator.wuanandroid.Bean.UserBean.User;
import com.example.administrator.wuanandroid.R;
import com.example.administrator.wuanandroid.localAPI.UserApi;

import com.example.administrator.wuanandroid.utils.RetrofitSingle;
import com.example.administrator.wuanandroid.utils.UserHelper;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class GroupingActivity extends AppCompatActivity implements View.OnClickListener {

    TextView webgroup, phpgroup,
            javagroup, uigroup,
            pmgroup, androidgroup,
            submitgroup;
    private int group_id;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_grouping);
        //initBundle();
        initView();
    }

   // private void initBundle() {
    //    Bundle bundle=getIntent().getBundleExtra("bundle");
    //    String name=bundle.getString("name");
    //}

    private void initView() {
        webgroup = findViewById(R.id.WebGroup);
        phpgroup = findViewById(R.id.PhpGroup);
        javagroup = findViewById(R.id.JavaGroup);
        uigroup = findViewById(R.id.UiGroup);
        pmgroup = findViewById(R.id.PmGroup);
        androidgroup = findViewById(R.id.AndroidGroup);
        submitgroup = findViewById(R.id.SubmitGroup);
        webgroup.setOnClickListener(this);
        phpgroup.setOnClickListener(this);
        javagroup.setOnClickListener(this);
        uigroup.setOnClickListener(this);
        pmgroup.setOnClickListener(this);
        androidgroup.setOnClickListener(this);
        submitgroup.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.WebGroup:
                WebGroup();
                break;
            case R.id.PhpGroup:
                PhpGroup();
                break;
            case R.id.JavaGroup:
                JavaGroup();
                break;
            case R.id.UiGroup:
                UiGroup();
                break;
            case R.id.PmGroup:
                PmGroup();
                break;
            case R.id.AndroidGroup:
                AndroidGroup();
                break;
            case R.id.SubmitGroup:
                SubmitGroup();
                break;
        }
    }

    private void WebGroup() {
        group_id = 1;
    }

    private void PhpGroup() {
        group_id = 2;
    }

    private void JavaGroup() {
        group_id = 3;
    }

    private void UiGroup() {
        group_id = 4;
    }

    private void PmGroup() {
        group_id = 5;
    }

    private void AndroidGroup() {
        group_id = 6;
    }

    private void SubmitGroup() {

        if (group_id == 0) {
            Toast.makeText(getApplicationContext(), "请选择分组", Toast.LENGTH_SHORT).show();

        } else netRequest();
    }

    private void netRequest() {
        RetrofitSingle
                .getRetrofit()
                .create(UserApi.class)
                .grouping(UserHelper.currentUser(getApplicationContext()).getUser_id(), group_id)
                .enqueue(new Callback<GroupingResponse>() {
                    @Override
                    public void onResponse(Call<GroupingResponse> call, Response<GroupingResponse> response) {
                        if (response.isSuccessful()) {
                            GroupingResponse groupingResponse = response.body();
                            if (groupingResponse != null) {
                                if (groupingResponse.getInfoCode() == 500) {
                                    Toast.makeText(getApplicationContext(), groupingResponse.getInfoText(), Toast.LENGTH_SHORT).show();
                                } else if (groupingResponse.getInfoCode() == 301) {

                                    User user = UserHelper.currentUser(getApplicationContext());
                                    user.setGroup_id(groupingResponse.getGroup_id());
                                    UserHelper.saveUser(getApplicationContext(), user);
                                    Toast.makeText(getApplicationContext(), groupingResponse.getInfoText(), Toast.LENGTH_SHORT).show();
                                    groupingToMain();
                                }
                            }
                        }
                    }

                    @Override
                    public void onFailure(Call<GroupingResponse> call, Throwable t) {
                        t.printStackTrace();
                        Toast.makeText(getApplicationContext(), "未响应，请检查网络连接", Toast.LENGTH_SHORT).show();

                    }
                });
    }

    private void groupingToMain(){
        Intent intent = new Intent(this, LeaveActivity.class);
        startActivity(intent);
        finish();
    }

}

File: app/src/main/java/com/example/administrator/wuanandroid/ui/GroupingActivity.kt
package com.example.administrator.wuanandroid.ui

import android.app.AlertDialog
import android.app.Dialog
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.StaggeredGridLayoutManager
import android.view.View
import com.example.administrator.wuanandroid.Adapter.GroupAdapter
import com.example.administrator.wuanandroid.Adapter.GroupItemClickListener
import com.example.administrator.wuanandroid.Bean.GroupBean.AllGroupBean
import com.example.administrator.wuanandroid.Bean.GroupBean.GroupRequest
import com.example.administrator.wuanandroid.Bean.GroupBean.GroupResponse
import com.example.administrator.wuanandroid.MainActivity
import com.example.administrator.wuanandroid.R
import com.example.administrator.wuanandroid.utils.*
import com.google.gson.Gson
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_grouping.*
import okhttp3.RequestBody
import android.content.DialogInterface




class GroupingActivity :AppCompatActivity(), View.OnClickListener,GroupItemClickListener {
    override fun ClickListner(num: Int) {

        val builder = AlertDialog.Builder(this@GroupingActivity)
        builder.setTitle("分组选择")
                .setMessage("您确定选择该分组吗")
                .setPositiveButton("确定", DialogInterface.OnClickListener { dialog, id ->
                    group_id = num
                    l.i(""+group_id)
                    netRequest()})
                .setNegativeButton("取消", DialogInterface.OnClickListener { dialog, id ->group_id = 0 })
        builder.create()
        builder.show()

    }

    private var group_id: Int = 0
    private var sharedUtil = SharedUtil()
    private var toast = ToastUtil(this)
    private var l = L()
    private var t = ToastUtil(this@GroupingActivity)
    var dialog :Dialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_grouping)
        getGroupList()

//        dialog = DialogUtil(this@GroupingActivity).getLoaddingDialog()
    }

    fun getGroupList(){
//        dialog!!.show()
//        dialog!!.setCancelable(false)
        RequestUtil.observable.getAllGroup()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe ({

                    AllGroupBean->
//                    dialog!!.dismiss()
                    var list = ArrayList<AllGroupBean.GroupsBean>()
                    l.i("执行了")
                    for (item in AllGroupBean.groups!!){list.add(item) }
                    var manager = StaggeredGridLayoutManager(2,LinearLayoutManager.VERTICAL)
                    var adapter = GroupAdapter(this@GroupingActivity,list,this)
                    group_recycler.layoutManager = manager
                    group_recycler.adapter = adapter

                },{
//                    dialog!!.dismiss()
                    t.st("发生${it.message}错误")
        })
    }

    override fun onClick(v: View) {
        when (v.id) {

        }
    }


    private fun netRequest() {
//        dialog!!.show()
        var gr = GroupRequest()
        gr.userId = sharedUtil.getInt(this@GroupingActivity,StaticClass.USER_ID,1)
        gr.groupId = group_id
        var gson = Gson()
        var strGr = gson.toJson(gr)
        val body = RequestBody.create(okhttp3.MediaType.parse("application/json; charset=utf-8"), strGr)
        RequestUtil.observable.IGroup(body)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe{
                    GroupResponse ->
//                    dialog!!.dismiss()
                    when(GroupResponse.infoCode){
                        200 -> {
                            sharedUtil.putInt(this@GroupingActivity,StaticClass.USER_ID, GroupResponse.userId!!)
                            sharedUtil.putInt(this@GroupingActivity,StaticClass.GROUP_ID,GroupResponse.groupId!!)
                            toast.lt(GroupResponse.infoText)
                            groupingToMain()
                        }
                        500 ->{toast.st(GroupResponse.infoText)}
                    }
                }

    }

    private fun groupingToMain() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }

}